tcms/testplans/static/testplans/js/mutable.js

```diff
@@ -10,6 +10,13 @@ $(document).ready((function () {
     $('#add_id_version').click(function () {
         return showRelatedObjectPopup(this);
     });
+
+    $('.selectpicker').selectpicker();
+
+    document.getElementById('id_product').onchange = function(){
+        $('.selectpicker').selectpicker('refresh');
+        populateProductVersion();
```

Comment:
This is correctly updating the Version select when a new Product is added. However it has the unwanted side effect that the first version is shown twice when changing between existing products. Test it more carefully next time.

Also:

1) The refresh action should be applied only to the product select
2) This new onchange handler duplicates the change handler so you have to remove `.change()`

3) Also missing the refresh for the Version select - again test it more carefully before submitting.

Reply:
the update for the Version select is done inside `populateProductVersion`

Reply:
Nope, it is not. When you + a new Version object the product field is not changed in any way and the Version select drop-down is not refreshed.